33-lifecycle-lab_v16/src/03-custominput.jsx

```diff
@@ -1,10 +1,80 @@
-/**
- * Implment an input component that shows 4
- * input boxes writing user input in the currently
- * focused panel
- *
- * See lab description on the webpage for a live example
- * how this should work.
- */
-
-
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class SquareFocus extends React.Component{
+
+    constructor(props) {
+        super(props);
+        this.state = {texts : []};
+    }
+
+
+    setFocus = (el,index) =>{
+        if(el!=null){
+            el.focus()
+            el.style.backgroundColor = "#ccc";
+        }
+    }
+
+    fillText = (e,index) =>{
+        let isFocused = (document.activeElement === e.target);
+        if(isFocused){
+            if(document.activeElement.innerHTML != ''){

```

Comment:
Don't change the `innerHTML` AND the `state`. Choose one. The React way is to change state, which leads to render, which will change the DOM.